File: Advanced Methods of Programming/Laborator1JAVA/src/main/java/service/ServiceAngajat.java
package service;

import domain.Angajat;
import repo.RepoAngajat;
import repo.RepoException;

public class ServiceAngajat {
    private RepoAngajat repoAngajat;

    public ServiceAngajat(RepoAngajat ra){
        this.repoAngajat=ra;
    }

    public void save(Angajat object) {
        this.repoAngajat.save(object);
    }

    public Iterable<Angajat> findAll() {
        return this.repoAngajat.findAll();
    }

    public Angajat findOne(String s) {
        Angajat angajat = this.repoAngajat.findOne(s);
        if(angajat==null)
            throw new RepoException("Angajat inexistent!");
        else return angajat;
    }

    public int size(){ return this.repoAngajat.size(); }
}

File: Advanced Methods of Programming/Lab1C-MPP/Lab1C-MPP/Controller/ServiceAngajat.cs
using Lab1C_MPP.Domain;
using Lab1C_MPP.Repo;
using System.Collections.Generic;

namespace Lab1C_MPP.Controller
{
    class ServiceAngajat
    {
        private RepoAngajat repoAngajat;

        public ServiceAngajat(RepoAngajat repo)
        {
            this.repoAngajat = repo;
        }

        public void Save(Angajat angajat)
        {
            this.repoAngajat.Save(angajat);
        }

        public int Size()
        {
            return this.repoAngajat.Size();
        }

        public Angajat FindOne(string id)
        {
            Angajat angajat = this.repoAngajat.FindOne(id);
            if (angajat == null) throw new RepositoryException("Angajat inexistent!");
            else return angajat;
        }

        public IEnumerable<Angajat> FindAll()
        {
            return this.repoAngajat.FindAll();
        }
    }
}
